src/SystemServicesContext/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { createContext, FunctionComponent, useContext } from "react"
+
+import useIsBluetoothOn from "./useIsBluetoothOn"
+import useIsLocationOn from "./useIsLocationOn"
+import useIsLocationRequired from "./useIsLocationRequired"
+
+export enum LocationPermissions {
+  NOT_REQUIRED,
+  REQUIRED_OFF,
+  REQUIRED_ON,
+}
+
+export interface SystemServicesState {
+  isBluetoothOn: boolean
+  locationPermissions: LocationPermissions
+}
+
+const initialState: SystemServicesState = {
+  isBluetoothOn: false,
+  locationPermissions: LocationPermissions.REQUIRED_OFF,
+}
+
+export const SystemServicesContext = createContext<SystemServicesState>(
+  initialState,
+)
+
+const SystemServicesProvider: FunctionComponent = ({ children }) => {
+  const isBluetoothOn = useIsBluetoothOn()
```

Comment:
nice and clean! I like how we have a clean separation between the Device State and the AppUI State and a function to convert from one to the other.

Reply:
Yeah agreed!